SKYKitChat/Classes/UI/ConversationView/SKYChatConversationViewController.swift

```diff
@@ -1691,10 +1691,39 @@ extension SKYChatConversationViewController {
                 print("Cannot start recording session.")
             }
         } else {
-            recordingSession.requestRecordPermission() { allowed in }
+            recordingSession.requestRecordPermission() { [weak self] allowed in
+                if !allowed {
+                    self?.showPermissionDeniedAlert()
+                }
+            }
         }
     }
 
+    open func showPermissionDeniedAlert() {
+        let alertController = UIAlertController(
+            title: "Unable to send voice messaage",
+            message: "Microphone permission is needed for sending voice messagae. " +
+                     "Please grant the permission in the application Settings.",
+            preferredStyle: .alert
+        )
+
+        let settingsAction = UIAlertAction(title: "Settings", style: .default) { (_) -> Void in
+            guard let settingsUrl = URL(string: UIApplicationOpenSettingsURLString) else {
+                return
+            }
+
+            if UIApplication.shared.canOpenURL(settingsUrl) {
```

Comment:
Just open the URL without checking.

Alternatively, remove the `Settings` button if the app cannot open the URL.